tests/e2e/pkg/tests/tests.go

```diff
@@ -0,0 +1,584 @@
+			for _, test := range tests {
+				test.Run()
+			}
+		})
+	}
+}
+
+func (tests TestPool) Sort() {
+	sort.SliceStable(tests, func(i, j int) bool {
+		return tests[i].less(tests[j])
+	})
+}
+
+func (tests TestPool) getSortedTestsByClusterID() map[string][]Test {
+	testsByClusterID := make(map[string][]Test)
+	// Need to be sorted to achieve test specs tree consistency between processes
+	// otherwise it can happen that specs order will be different for each process
+	tests.Sort()
+
+	for _, test := range tests {
+		testsByClusterID[test.k8sCluster.clusterInfo.clusterID] = append(testsByClusterID[test.k8sCluster.clusterInfo.clusterID], test)
+	}
+
+	return testsByClusterID
+}
+
+func (tests TestPool) getTestsByContextName() map[string][]Test {
+	testsByContextName := make(map[string][]Test)
+	for _, test := range tests {
+		testsByContextName[test.k8sCluster.kubectlOptions.ContextName] = append(testsByContextName[test.k8sCluster.kubectlOptions.ContextName], test)
+	}
+
+	return testsByContextName
+}
+
+func (tests TestPool) getTestsByVersions() map[string][]Test {
+	testsByVersions := make(map[string][]Test)
+	for _, test := range tests {
+		testsByVersions[test.k8sCluster.clusterInfo.version] = append(testsByVersions[test.k8sCluster.clusterInfo.version], test)
+	}
+
+	return testsByVersions
+}
+
+func (tests TestPool) getTestsByProviders() map[string][]Test {
+	testsByProviders := make(map[string][]Test)
+	for _, test := range tests {
+		testsByProviders[test.k8sCluster.clusterInfo.provider] = append(testsByProviders[test.k8sCluster.clusterInfo.provider], test)
+	}
+
+	return testsByProviders
+}
+
+// GetTestSuiteDurationParallel calculates the expected duration of the tests when
+// suite is executed in parallel.
+func (tests TestPool) GetTestSuiteDurationParallel() time.Duration {
+	testsByClusterID := tests.getSortedTestsByClusterID()
+
+	var max time.Duration
+	for _, tests := range testsByClusterID {
+		var localDuration time.Duration
+		for _, test := range tests {
+			localDuration += test.testCase.Duration
+		}
+
+		if localDuration > max {
+			max = localDuration
+		}
+	}
+	return max
+}
+
+// GetTestSuiteDurationSerial calculates the expected duration of the tests when
+// test suite is executed in serial.
+func (tests TestPool) GetTestSuiteDurationSerial() time.Duration {
+	var allDuration time.Duration
+	for _, test := range tests {
+		allDuration += test.testCase.Duration
+	}
+	return allDuration
+}
+
+// GetTestSuiteSpecsCount returns the number of specs in the pool based on the testCase explicit information.
+func (tests TestPool) GetTestSuiteSpecsCount() int {
+	var specsCount int
+	for _, test := range tests {
+		specsCount += test.testCase.SpecsCount
+	}
+	return specsCount
+}
+
+// GetParallelTotal returns the maximum number of the possible parallelization
+func (tests TestPool) GetParallelTotal() int {
+	return len(maps.Keys(tests.getSortedTestsByClusterID()))
+}
```

Comment:
Question: Isn't this just the number of clusters? Do we need a function for this?